src/Component/Filter/FilterTree/FilterTree.tsx

```diff
@@ -0,0 +1,567 @@
+
+    let addedFilter: GsFilter ;
+    const newFilter: GsFilter = [...filter];
+
+    switch (type) {
+      case 'and':
+        addedFilter = ['&&', ['==', '', ''], ['==', '', '']];
+        break;
+      case 'or':
+        addedFilter = ['||', ['==', '', ''], ['==', '', '']];
+        break;
+      case 'not':
+        addedFilter = ['!', ['==', '', '']];
+        break;
+      case 'comparison':
+      default:
+        addedFilter = ['==', '', ''];
+        break;
+    }
+
+    if (position === '') {
+      newFilter.push(addedFilter);
+    } else {
+      const previousFilter = _get(newFilter, position);
+      previousFilter.push(addedFilter);
+      _set(newFilter, position, previousFilter);
+    }
+
+    onFilterChange(newFilter);
+  }
+
+  /**
+   * Changes a filter at a position to a given typ.
+   *
+   */
+  changeFilter = (position: string, type: string) => {
+    const {
+      filter,
+      onFilterChange
+    } = this.props;
+
+    let addedFilter: GsFilter ;
+    const newFilter: GsFilter = [...filter];
+    const previousFilter = position === '' ? newFilter : _get(newFilter, position);
+
+    switch (type) {
+      case 'and':
+        if (previousFilter && previousFilter[0] === '&&' || previousFilter[0] === '||' ) {
+          addedFilter = previousFilter;
+          addedFilter[0] = '&&';
+        } else {
+          addedFilter = ['&&', ['==', '', ''], ['==', '', '']];
+        }
+        break;
+      case 'or':
+        if (previousFilter && previousFilter[0] === '&&' || previousFilter[0] === '||' ) {
+          addedFilter = previousFilter;
+          addedFilter[0] = '||';
+        } else {
+          addedFilter = ['||', ['==', '', ''], ['==', '', '']];
+        }
+        break;
+      case 'not':
+        addedFilter = ['!', ['==', '', '']];
+        break;
+      case 'comparison':
+      default:
+        addedFilter = ['==', '', ''];
+        break;
+    }
+
+    if (position === '') {
+      onFilterChange(addedFilter);
+      return;
+    } else {
+      _set(newFilter, position, addedFilter);
+    }
+
+    onFilterChange(newFilter);
+  }
+
+  /**
+   * Removes a filter at a given position.
+   *
+   */
+  removeFilter = (position: string) => {
+    const {
+      filter,
+      onFilterChange
+    } = this.props;
+
+    const parentPosition = position.substring(0, position.length - 3);
+    const parentFilter = this.getFilterAtPosition(parentPosition);
+    let newFilter;
+
+    if (parentFilter.length <= 2) {
+      newFilter = this.removeAtPosition(filter, parentPosition);
+    } else {
+      newFilter = this.removeAtPosition(filter, position);
+    }
+
+    onFilterChange(newFilter);
+  }
+
+  /**
+   * Transforms a position String like '[2][3]' to an positionArray like [2, 3].
+   */
+  positionStringAsArray = (positionString: string) => {
+    return positionString
+      .replace('][', ',')
+      .replace(']', '')
+      .replace('[', '')
+      .split(',')
+      .map(i => parseInt(i, 10));
```

Comment:
This will break if using more than 9 filters, because the map function will parse over each digit instead of each number.
`'[10][5]' => [1,0,5]`

Reply:
Line 387 replaces `][` with `,`
Line 388 removes `[`
Line 389 removes `]`
Line 390 splits the string by the added `,` from Line 387

So `'[10][3]'` becomes `["10", "3"]` to finally become `[10, 3]`.